src/test/java/com/pusher/client/channel/impl/PrivateEncryptedChannelImplTest.java

```diff
@@ -0,0 +1,173 @@
+    public void testPresenceChannelName() {
+        newInstance("presence-stuffchannel");
+    }
+
+    @Override
+    @Test
+    public void testPrivateEncryptedChannelName() {
+        newInstance("private-encrypted-stuffchannel");
+    }
+
+    @Override
+    @Test(expected = IllegalArgumentException.class)
+    public void testPrivateChannelName() {  newInstance("private-stuffchannel");  }
+
+    /*
+    TESTING SUBSCRIBE MESSAGE
+     */
+
+    @Override
+    @Test
+    public void testReturnsCorrectSubscribeMessage() {
+        assertEquals("{\"event\":\"pusher:subscribe\",\"data\":{" +
+                "\"channel\":\"" + getChannelName() + "\"," +
+                "\"auth\":\"636a81ba7e7b15725c00:3ee04892514e8a669dc5d30267221f16727596688894712cad305986e6fc0f3c\""+
+                "}}", channel.toSubscribeMessage());
+    }
+
+    /*
+    TESTING AUTHENTICATION METHOD
+     */
+
+    @Test
+    public void checkAuthenticationSucceedsGivenValidAuthorizer() {
+        when(mockAuthorizer.authorize(Matchers.anyString(), Matchers.anyString()))
+                .thenReturn(authorizer_valid);
+
+        PrivateEncryptedChannelImpl channel = new PrivateEncryptedChannelImpl(
+                mockInternalConnection, getChannelName(), mockAuthorizer, mockedFactory);
+
+        channel.checkAuthentication();
+    }
+
+    protected ChannelEventListener getEventListener() {
+        return mock(PrivateEncryptedChannelEventListener.class);
+    }
+
+    @Test(expected = AuthorizationFailureException.class)
+    public void checkAuthenticationThrowsExceptionIfNoAuthKey() {
+        when(mockAuthorizer.authorize(Matchers.anyString(), Matchers.anyString()))
+                .thenReturn(authorizer_missingAuthKey);
+
+        PrivateEncryptedChannelImpl channel = new PrivateEncryptedChannelImpl(
+                mockInternalConnection, getChannelName(), mockAuthorizer, mockedFactory);
+
+        channel.checkAuthentication();
+    }
+
+    @Test(expected = AuthorizationFailureException.class)
+    public void checkAuthenticationThrowsExceptionIfNoSharedSecret() {
+        when(mockAuthorizer.authorize(Matchers.anyString(), Matchers.anyString()))
+                .thenReturn(authorizer_missingSharedSecret);
+
+        PrivateEncryptedChannelImpl channel = new PrivateEncryptedChannelImpl(
+                mockInternalConnection, getChannelName(), mockAuthorizer, mockedFactory);
+
+        channel.checkAuthentication();
+    }
+
+    @Test(expected = AuthorizationFailureException.class)
+    public void checkAuthenticationThrowsExceptionIfMalformedJson() {
+        when(mockAuthorizer.authorize(Matchers.anyString(), Matchers.anyString()))
+                .thenReturn(authorizer_malformedJson);
+
+        PrivateEncryptedChannelImpl channel = new PrivateEncryptedChannelImpl(
+                mockInternalConnection, getChannelName(), mockAuthorizer, mockedFactory);
+
+        channel.checkAuthentication();
+    }
+
+    /*
+    TESTING SECRET BOX
```

Comment:
I don't necessarily thing such headers are needed, but if really they are it seem like an indication that it would be good to extract such a part to a separate well named test class.

Reply:
The headers are for me to try to chunk things together to work out how we might refactor these in future as I found it difficult to know which tests I needed to override and add to be covered :-)